Eastern-Research-Group/mywaterway
# Feature/json file for web services

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3529154

## Main Changes:
* Added the `config/services.json` lookup file, so that services can be updated without changing the code. 
* Updated the google analytics logic to use the services.json file. 
* I ended up using context to hold the service urls and a window variable to hold the google analytics logging. I figured it was best to keep the service urls out of the window object. Putting the google analytics map in the window object made it easier to handle that lookup file.
* I put a global error message in to handle the services lookup file not loading. This file is needed for pretty much everything in the app. 

## Steps To Test:
1. I recommend adding this line ```console.log(`eventAction: "${eventAction}" url: "${url}"`);``` after line 141 in utils/fetchUtils.js. This will make it easier to tell if the google analytics mapping logic is working. You should not see any "UNKNOWN" event actions.
2. Open the dev console. 
3. For every page in the app, perform the following steps:
    1. Open the page.
    2. Verify that all of the web services work correctly. 
        Note: Keep in mind that some things need to be expanded to test additional service calls (i.e. Documents section on the state page, monitoring location popups, etc.). 
    3. Verify none of the eventActions in the console are "UNKNOWN". 
    4. Block the `http://localhost:9091/proxy?url=http://localhost:9091/data/config/services.json` service call.
    5. Verify the "How's My Waterway is currently unavailable" error is displayed.
